pkg/querier/batch/batch.go

```diff
@@ -129,25 +129,39 @@ func (a *iteratorAdapter) Next() chunkenc.ValueType {
 	}
 
 	if a.curr.Index < a.curr.Length {
-		// TODO native histograms: return correct batch type
-		return chunkenc.ValFloat
+		return a.curr.ValueType
 	}
 	return chunkenc.ValNone
 }
 
 // At implements chunkenc.Iterator.
 func (a *iteratorAdapter) At() (int64, float64) {
+	if a.curr.ValueType != chunkenc.ValFloat {
+		panic(fmt.Sprintf("Cannot read float from batch %v", a.curr.ValueType))
```

Comment:
@codesome and me think panic() in iterator is justified as it would indicate coding error (user not using switch to select function) and not data error, these should be caught at testing. Adding `error` return value is costly. Changing to returning 0 value  would violate our rule to either return correct results or none at all.